inbox/events/microsoft/parse.py

```diff
@@ -81,6 +81,16 @@ def parse_msgraph_datetime_tz_as_utc(datetime_tz: MsGraphDateTimeTimeZone):
     # so we need to use ciso8601 because Python stdlib only supports up to 6.
     dt = ciso8601.parse_datetime(datetime_tz["dateTime"])
 
+    # Microsoft sometimes returns payloads such as this one:
+    # {"dateTime": "0001-01-01T00:00:00.0000000Z", "timeZone": "tzone://Microsoft/Utc"}
+    # Notice the `Z` - the `dateTime` is timezone-aware. This is undocumented
+    # behaviour, but we will handle it here so it doesn't break sync.
+    if dt.tzinfo:
```

Comment:
should we just instead honor the tzinfo from dateTime always if it's set?

Reply:
We have no evidence of Microsoft returning values other than `0001-01-01T00:00:00.0000000Z`, so I'd like to minimise the changes to the logic that we do here, and only handle the UTC use-case.

I think it's better to fail for something we don't expect here (which we do already anyway), than to store an incorrectly calculated value by accident.

What do you think?